### PR TITLE
Eq5 203 monitor get signed contracts

### PR DESCRIPTION
**Pull request by OussaKab**

### CODE REVIEW

File: backend/src/main/java/com/gestionnaire_de_stage/controller/ContractController.java [hunk 1]
```diff
@@ -141,7 +141,7 @@ public ContractController(ContractService contractService, TemplateEngine templa
     }
 
     @GetMapping("/student/{student_id}")//SESSION : get only contract of current or futur session
-    public ResponseEntity<?> ContractNeedsStudentSignature(@PathVariable Long student_id) {
+    public ResponseEntity<?> getContractByStudentId(@PathVariable Long student_id) {
```
Comment (BertSa) [inline backend/src/main/java/com/gestionnaire_de_stage/controller/ContractController.java:144]:
```suggestion
    public ResponseEntity<?> ContractNeedsStudentSignature(@PathVariable Long student_id) {
```

File: backend/src/test/java/com/gestionnaire_de_stage/controller/ContractControllerTest.java [hunk 2]
```diff
@@ -404,6 +404,78 @@ public void testSignContractPDFStudent_withInvalidEntries() throws Exception {
...
+        MvcResult mvcResult = mockMvc.perform(
+            MockMvcRequestBuilders.get("/contracts/manager/signed/" + nonExistentId)
+                .contentType(MediaType.APPLICATION_JSON))
+        .andReturn();
+
+        final MockHttpServletResponse response = mvcResult.getResponse();
+        assertThat(response.getStatus()).isEqualTo(HttpStatus.OK.value());
+
+        List<Contract> returnedContracts = MAPPER.readValue(response.getContentAsString(), new TypeReference<>() {});
+        assertThat(returnedContracts).isEmpty();
```
Comment (BertSa) [inline backend/src/test/java/com/gestionnaire_de_stage/controller/ContractControllerTest.java:440]:
```suggestion
        long nonExistentId = 1000L;
        when(contractService.getAllSignedContractsByManager(any())).thenReturn(Collections.emptyList());
        
        MvcResult mvcResult = mockMvc.perform(
            MockMvcRequestBuilders.get("/contracts/manager/signed/" + nonExistentId)
                .contentType(MediaType.APPLICATION_JSON))
        .andReturn();

        final MockHttpServletResponse response = mvcResult.getResponse();
        List<Contract> returnedContracts = MAPPER.readValue(response.getContentAsString(), new TypeReference<>() {});
        assertThat(response.getStatus()).isEqualTo(HttpStatus.OK.value());
        assertThat(returnedContracts).isEmpty();
```

File: backend/src/test/java/com/gestionnaire_de_stage/controller/ContractControllerTest.java [hunk 3]
```diff
@@ -404,6 +404,78 @@ public void testSignContractPDFStudent_withInvalidEntries() throws Exception {
...
+        .andReturn();
+
+        final MockHttpServletResponse response = mvcResult.getResponse();
+        assertThat(response.getStatus()).isEqualTo(HttpStatus.OK.value());
+
+        List<Contract> returnedContracts = MAPPER.readValue(response.getContentAsString(), new TypeReference<>() {});
+
+        assertThat(returnedContracts)
+                .isNotEmpty()
+                .isEqualTo(dummyContracts);
```
Comment (BertSa) [inline backend/src/test/java/com/gestionnaire_de_stage/controller/ContractControllerTest.java:423]:
```suggestion
        List<Contract> dummyContracts = getDummyContractList();
        when(contractService.getAllSignedContractsByManager(any())).thenReturn(dummyContracts);
        
        MvcResult mvcResult = mockMvc.perform(
            MockMvcRequestBuilders.get("/contracts/manager/signed/" + 100L)
                .contentType(MediaType.APPLICATION_JSON))
        .andReturn();

        final MockHttpServletResponse response = mvcResult.getResponse();
        List<Contract> returnedContracts = MAPPER.readValue(response.getContentAsString(), new TypeReference<>() {});
        assertThat(response.getStatus()).isEqualTo(HttpStatus.OK.value());
        assertThat(returnedContracts)
                .isNotEmpty()
                .isEqualTo(dummyContracts);
```

File: backend/src/test/java/com/gestionnaire_de_stage/controller/ContractControllerTest.java [hunk 4]
```diff
@@ -404,6 +404,78 @@ public void testSignContractPDFStudent_withInvalidEntries() throws Exception {
...
+                .andReturn();
+
+        final MockHttpServletResponse response = mvcResult.getResponse();
+
+        List<Contract> returnedContracts = MAPPER.readValue(response.getContentAsString(), new TypeReference<>() {});
+
+        assertThat(response.getStatus()).isEqualTo(HttpStatus.OK.value());
+        assertThat(returnedContracts)
+                .isNotEmpty()
+                .isEqualTo(dummyContracts);
```
Comment (BertSa) [inline backend/src/test/java/com/gestionnaire_de_stage/controller/ContractControllerTest.java:459]:
```suggestion
        List<Contract> dummyContracts = getDummyContractList();
        when(contractService.getAllSignedContractsByMonitor(any())).thenReturn(dummyContracts);
        
        MvcResult mvcResult = mockMvc.perform(
                        MockMvcRequestBuilders.get("/contracts/monitor/signed/" + 100L)
                                .contentType(MediaType.APPLICATION_JSON))
                .andReturn();

        final MockHttpServletResponse response = mvcResult.getResponse();
        List<Contract> returnedContracts = MAPPER.readValue(response.getContentAsString(), new TypeReference<>() {});
        assertThat(response.getStatus()).isEqualTo(HttpStatus.OK.value());
        assertThat(returnedContracts)
                .isNotEmpty()
                .isEqualTo(dummyContracts);
```

File: backend/src/test/java/com/gestionnaire_de_stage/controller/ContractControllerTest.java [hunk 5]
```diff
@@ -404,6 +404,78 @@ public void testSignContractPDFStudent_withInvalidEntries() throws Exception {
...
+        MvcResult mvcResult = mockMvc.perform(
+                        MockMvcRequestBuilders.get("/contracts/monitor/signed/" + nonExistentId)
+                                .contentType(MediaType.APPLICATION_JSON))
+                .andReturn();
+
+        final MockHttpServletResponse response = mvcResult.getResponse();
+        assertThat(response.getStatus()).isEqualTo(HttpStatus.OK.value());
+
+        List<Contract> returnedContracts = MAPPER.readValue(response.getContentAsString(), new TypeReference<>() {});
+        assertThat(returnedContracts).isEmpty();
```
Comment (BertSa) [inline backend/src/test/java/com/gestionnaire_de_stage/controller/ContractControllerTest.java:476]:
```suggestion
        long nonExistentId = 1000L;
        when(contractService.getAllSignedContractsByManager(any())).thenReturn(Collections.emptyList());

        MvcResult mvcResult = mockMvc.perform(
                        MockMvcRequestBuilders.get("/contracts/monitor/signed/" + nonExistentId)
                                .contentType(MediaType.APPLICATION_JSON))
                .andReturn();

        final MockHttpServletResponse response = mvcResult.getResponse();
        List<Contract> returnedContracts = MAPPER.readValue(response.getContentAsString(), new TypeReference<>() {});
        assertThat(response.getStatus()).isEqualTo(HttpStatus.OK.value());
        assertThat(returnedContracts).isEmpty();
```

File: gestionnaire_de_stage_frontend/src/components/ViewSignedContracts/ViewSignedContract/ViewSignedContract.jsx [hunk 6]
```diff
@@ -0,0 +1,28 @@
...
+import {useEffect, useState} from "react";
+import {toast} from "../../../utility";
+import {FiDownload} from "react-icons/all";
+
+export default function ViewSignedContract({contract}) {
+    const [pdf, setPdf] = useState(null);
+
+    useEffect(() => setPdf(toPdfBlob(contract.contractPDF)), [contract.contractPDF]);
+
+    const _studentFullName = student => `${student.firstName} ${student.lastName}`;
```
Comment (BertSa) [inline gestionnaire_de_stage_frontend/src/components/ViewSignedContracts/ViewSignedContract/ViewSignedContract.jsx:12]: pas necessaire
```suggestion
    const studentFullName = student => `${student.firstName} ${student.lastName}`;
```

**Review by BertSa — CHANGES_REQUESTED** [review]
Petit changement a faire mais sinon guddo jobu!

**Comment by Ishnewalon** [comment]
Work seems good

**Review by Ishnewalon — COMMENTED** [review]
Seems good enough

**Review by BertSa — APPROVED** [review]

**Review by Ishnewalon — APPROVED** [review]
Seems legit

**Pull request merged**